### PR TITLE
For each

### PR DESCRIPTION
**Pull request by agrazda**

### CODE REVIEW

**Comment by agrazda** [comment]
completed 3 and tried for 4

File: javascript/code-challenges/challenges-01.test.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+Write a function named `allUpperCase` that takes an array of strings, and returns a new array of the strings converted to upper case.
+
+Use `forEach` to loop over the input array. The modified strings should each be added into a local array. Return that local array.
+------------------------------------------------------------------------------------------------ */
+
+const allUpperCase = (arr) => {
+  const newArray = []
+  // Solution code here...
+
+  arr.forEach(caps => {
```
Comment (RivaD2) [inline javascript/code-challenges/challenges-01.test.js:55]: Nice work so far here! 

I wanted to provide you some food for thought regarding variable names. No need to change the code but for this challenge, what are some other names we could have given `caps`? What type of value is `caps`

File: javascript/code-challenges/challenges-01.test.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+Then, write a function named `speaker` that takes in an array of strings and a callback function. 
+
+Use `forEach` to build a new array of strings, each string modified by the callback. Return the new array. 
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => `${word.toUpperCase()}!`
+  // Solution code here...
+
+ 
+const speaker = (words, callback) => {
```
Comment (RivaD2) [inline javascript/code-challenges/challenges-01.test.js:77]: Hey there, nice work so far! 👍 
 You are getting closer, take  another look at what you are adding in front of the forEach and in front of `push().
 
   In other words, what do you need to iterate over?  Is it speaker or something else perhaps?

For push, what data structure to you want to push results into?

File: javascript/code-challenges/challenges-01.test.js [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr, value) => {
+  // Solution code here...
```
Comment (RivaD2) [inline javascript/code-challenges/challenges-01.test.js:106]: For this one, how can you get the value into the array? Can you push it? 👍

**Pull request merged**